tests/frameworks/test_lightning.py

```diff
@@ -350,3 +350,53 @@ def test_dvclive_logger_init_args():
             "dir": "results",  # Resolve from Live.__init__
         },
     )
+
+
+def log_metrics_from_outside_lightning(live):
+    step = 0
+    while True:
+        if live.step == step:
+            live.log_metric("test", 0.5)
+            step += 1
+        else:
+            time.sleep(0.001)
+
+
+@pytest.mark.timeout(3)
+def test_metrics_from_outside_lightning_updates_to_studio(
+    tmp_dir, mocked_dvc_repo, mocked_studio_post
+):
+    mocked_post, _ = mocked_studio_post
+
+    model = ValLitXOR()
+    dvclive_logger = DVCLiveLogger()
+    trainer = Trainer(
+        logger=dvclive_logger,
+        max_steps=4,
+        val_check_interval=2,
+        log_every_n_steps=1,
+        enable_checkpointing=False,
+    )
+
+    independent_logger = threading.Thread(
```

Comment:
do we step the tread anywhere?

Reply:
I'm guessing your meaning "stop". If that is the case, the thread runs as a daemon, which means that when Python ends execution, it stops the thread by itself :)

Reply:
It happens at the end of all the tests. In the meanwhile the thread is still active.
You're right it would have been a problem for the following tests.
I think we will change the implementation with the one @dberenbaum wrote in #790 as I find it cleaner.